Libraries/Lists/FlatList.js

```diff
@@ -518,8 +520,13 @@ class FlatList<ItemT> extends React.PureComponent<Props<ItemT>, void> {
     }
   };
 
-  _getItemCount = (data: ?Array<ItemT>): number => {
-    if (Array.isArray(data)) {
+  _getItemCount = (data: ?$ArrayLike<ItemT>): number => {
+    if (
+      data &&
+      (typeof data === 'object' || typeof data === 'function') &&
+      typeof data.length === 'number' &&
+      Symbol.iterator in data
```

Comment:
Is this just to get _some_ proof that it's array-like? I mean, neither the block below nor any other part of the component depends on the `data` being iterable.

Reply:
Yeah, Flow `$ArrayLike` requires iterability, where TypeScript `ArrayLike` does not, and I wanted to preemptively avoid the confusion the mismatch would cause and flag anything following the wrong `ArrayLike`.

But this actually ended up flagging that Flow `$ArrayLike` requiring iterability is more likely a mistake, and that requirement was added after-the-fact in a separate refactor. So I think it is likely this check will be removed.